test.js

```diff
@@ -290,7 +290,7 @@ test('safely handle missing package.json', t => {
 
 	let config;
 	t.notThrows(() => {
-		config = new Conf({projectName: 'conf-fixture-project-name'});
+		config = new Conf({projectName: 'conf-fixture-project-name', projectVersion: '1.0.0'});
```

Comment:
I think this change is unnecessary since the migration option is not being specified.

Reply:
But it now breaks, as it's not "secured" under the protective code of `if (!options.projectName)` on line 53.

Reply:
Hmm, I referring to not having to specify the projectVersion in this line, effectively removing the change. Or maybe I’m missing something?

Reply:
We should avoid breaking the current usage of the module.

Line 61 condition should only be true if we set the `migration` option. We can improve that conditional logic, in order to prevent this.

Reply:
You're both absolutely right (my brain was probably asleep last night).

I added a validation for the existance of  `packageData` (line 53 and 61), which was causing the compatibility to break.

Sorry about that 👍